.github/workflows/comprehensive-testing.yml

```
@@ -55,16 +55,18 @@ jobs:
 
       - name: Audit dependencies
-        run: npm audit --audit-level moderate
+        run: npm audit --audit-level critical
 
       - name: Run security tests
         run: npm run test:security
 
-      - name: CodeQL Analysis
-        uses: github/codeql-action/analyze@v2
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
```

actionlint (1.7.12)

[error] the runner of "github/codeql-action/init@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)